x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativeRolesStore.java

```diff
@@ -211,9 +221,18 @@ public void onFailure(Exception e) {
     public void putRole(final PutRoleRequest request, final RoleDescriptor role, final ActionListener<Boolean> listener) {
         if (role.isUsingDocumentOrFieldLevelSecurity() && DOCUMENT_LEVEL_SECURITY_FEATURE.checkWithoutTracking(licenseState) == false) {
             listener.onFailure(LicenseUtils.newComplianceException("field and document level security"));
-        } else {
-            innerPutRole(request, role, listener);
-        }
+        } else if (role.hasRemoteIndicesPrivileges()
+            && clusterService.state().nodes().getMinNodeVersion().before(RoleDescriptor.VERSION_REMOTE_INDICES)) {
```

Comment:
just double checking that this in only ever called from a user's action to add/update a role, right ? (if not then we should be really careful about which workflows throw during mixed clusters)

Reply:
yup this is only ever called via `TransportPutRoleAction` which in turns is only ever invoked through PUT role API calls

Reply:
What workflow do you have in mind? Internally (ILM, ML), I don't think we have anything that mutates security entities. Externally, creating and granting API keys are probably the most used security operations by stack and solutions. API keys will be supporting remote indices, so theorectically it could be an issue. Enterprise search also manages user roles. We should communicate this clearly to other teams, but I don't think it should not affect our decision here.